src/main/java/org/mule/extension/runtime/ExpirationPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.mule.extension.runtime;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A policy around when should a given instances be considered elegible for expiration.
+ * <p/>
+ * Notice that this contract is not directly tied to the instance to be expired itself. It's tied
+ * to the concept of idle based expiration.
+ *
+ * @since 1.0
+ */
+public interface ExpirationPolicy
+{
+
+    /**
+     * Determines if an instance should be expired based on the last time it was used.
```

Comment:
it's restrictive indeed. We thought about it with LG and we think that a config should never be expired based on any of those criterias because if it's in use you would have to recreate it anyway. The only case in which expiration makes sense is idleness. 

Thus, I don't think we should overdesign to accomodate other cases. If that comes to happen, I'm happy to refactor